### PR TITLE
SC 63.2 -- Transaction storing and accessing in history

### PR DESCRIPTION
**Pull request by oaleksandrov-ip**

### CODE REVIEW

File: sdk/src/main/scala/com/horizen/SidechainHistory.scala [hunk 1]
```diff
@@ -392,84 +394,71 @@ class SidechainHistory private (val storage: SidechainHistoryStorage,
...
   }
 
-  override def searchTransactionInsideSidechainBlock(transactionId: String, blockId: String): JOptional[Transaction] = {
-    storage.blockById(ModifierId(blockId)) match {
-      case Some(scBlock) => findTransactionInsideBlock(transactionId, scBlock)
-      case None => JOptional.empty()
-    }
+
+  private def findTransactionInBlock(transactionId: ModifierId, block: SidechainBlock) : Option[Transaction] = {
+    block.transactions.find(box => box.id.equals(transactionId))
```
Comment (SergeyMakukha) [inline sdk/src/main/scala/com/horizen/SidechainHistory.scala]: Maybe we need replace box by transaction? Current naming is confusable.

File: sdk/src/main/scala/com/horizen/SidechainApp.scala [hunk 2]
```diff
@@ -127,19 +128,23 @@ class SidechainApp @Inject()
     //openStorage(new JFile(s"${sidechainSettings.scorexSettings.dataDir.getAbsolutePath}/state")),
     registerStorage(stateStorage),
     sidechainBoxesCompanion)
-  protected val sidechainHistoryStorage = new SidechainHistoryStorage(
+  protected val sidechainHistoryStorage = new SidechainBlocks(
```
Comment (SergeyMakukha) [inline sdk/src/main/scala/com/horizen/SidechainApp.scala]: Maybe rename SidechainBlocks to SidechainHistoryBlocksStorage?

File: sdk/src/main/scala/com/horizen/SidechainApp.scala [hunk 3]
```diff
@@ -127,19 +128,23 @@ class SidechainApp @Inject()
...
     //openStorage(new JFile(s"${sidechainSettings.scorexSettings.dataDir.getAbsolutePath}/history")),
     registerStorage(historyStorage),
     sidechainTransactionsCompanion, params)
-
+  protected val transactionIndexes = new TransactionIndexes(
```
Comment (SergeyMakukha) [inline sdk/src/main/scala/com/horizen/SidechainApp.scala]: Maybe rename TransactionIndexes to SidechainHistoryTransactionStorage?

File: sdk/src/main/scala/com/horizen/SidechainHistory.scala [hunk 4]
```diff
@@ -392,84 +394,71 @@ class SidechainHistory private (val storage: SidechainHistoryStorage,
...
   }
 
-  private def findTransactionInsideBlock(transactionId : String, block : SidechainBlock) : JOptional[Transaction] = {
-    block.transactions.find(box => box.id.equals(ModifierId(transactionId))) match {
-      case Some(tx) => JOptional.ofNullable(tx)
-      case None => JOptional.empty()
-    }
+  private def findTransactionInBlockId(transactionId: ModifierId, blockId: ModifierId): Option[Transaction] = {
```
Comment (SergeyMakukha) [inline sdk/src/main/scala/com/horizen/SidechainHistory.scala]: Maybe rename to findTransactionInBlockById

**Comment by i-Alex** [comment]
PR is outdated

**Pull request closed without merging**